src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/AnonymousTypeManager.Templates.cs

```diff
@@ -314,6 +328,15 @@ public void AssignTemplatesNamesAndCompile(MethodCompiler compiler, PEModuleBuil
 
             builder.Free();
 
+            // Ensure all previous sythesized delegates are included so the
+            // types are available for subsequent edit and continue generations.
+            foreach (var key in moduleBeingBuilt.GetPreviousSynthesizedDelegates())
+            {
+                GeneratedNames.ParseDynamicCallSiteName(key.Name, out var refKinds, out var returnsVoid, out var generation, out var parameterCount);
```

Comment:
> GeneratedNames.ParseDynamicCallSiteName(key.Name, out var refKinds, out var returnsVoid, out var generation, out var parameterCount); [](http://example.com/codeflow?start=15&length=134)

The name is parsed here and then again in CreatePlaceholderSynthesizedDelegateValue. Can the key store the parsed values instead of the name?